project_task_personal_stage_auto_fold/models/project_task.py

```diff
@@ -0,0 +1,42 @@
+from odoo import models
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+
+    def write(self, vals):
+        result = super().write(vals)
+        if "stage_id" in vals and vals.get("stage_id"):
```

Comment:
This can be simplified with:

```suggestion
        if vals.get("stage_id"):
```